fix(i18n): fallback SSR (#10755)
withastro/astro: .changeset/old-pugs-jog.md

---
"astro": patch
---

> Fixes a case where the i18n fallback faild to correctly redirect to the index when the fallback is enabled in SSR

Reviewer: ```suggestion
Fixes a case where the i18n fallback failed to correctly redirect to the index page with SSR enabled
```
